src/utils/useStickyTabs.ts

```diff
@@ -0,0 +1,75 @@
+import { useCallback, useEffect, useState } from 'react';
+import { useHistory } from 'react-router-dom';
+
+import { Tab } from '@terraware/web-components';
+
+import useQuery from './useQuery';
+import useStateLocation, { getLocation } from './useStateLocation';
+
+interface StickyTabsProps {
+  defaultTab: string;
+  tabs: Tab[];
+  viewIdentifier: string;
+}
+
+const makeTabSessionKey = (viewIdentifier: string) => `tab-${viewIdentifier}`;
+
+const getTabFromSession = (viewIdentifier: string): string => {
+  try {
+    return sessionStorage.getItem(makeTabSessionKey(viewIdentifier)) || '';
+  } catch (e) {
+    return '';
+  }
+};
+
+const writeTabToSession = (viewIdentifier: string, tab: string): void => {
+  try {
+    sessionStorage.setItem(makeTabSessionKey(viewIdentifier), tab);
+    // tslint:disable-next-line:no-empty
+  } catch (e) {}
```

Comment:
We don't really care if session storage is inaccessible, falls back to default tab behavior.